### PR TITLE
add gulp task runner

### PR DESCRIPTION
**Pull request by cmuecke**
@matthistuff ready for review.

### CODE REVIEW

**Comment by matthistuff** [comment]
Please update `Dockerfile` and the scripts in `package.json`. We probably do not need any of the scripts anymore. You will need to put the gulp stuff in the `package.json` into the `dependencies` object as the server does run in `NODE_ENV=production` mode.

Can we add watch for browserify and styles?

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    return gulp.src('source/less/main.less')
+        .pipe(less({
+            onError: console.error.bind(console, 'Less error:')
+        }))
+        .pipe(gulp.dest('client/css/'));
+});
+
+gulp.task('browserify', function () {
+    return browserify({
+        entries: ['./source/js/client.js'],
```
Comment (matthistuff) [inline gulpfile.js]: Adding

``` js
        debug: process.env.NODE_ENV === 'development',
```

will give us source maps.

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    del = require('del');
+
+var options = {
+        continueOnError: true,
+        pipeStdout: true
+    };
+
+gulp.task('clean', del.bind(null, ['client']));
+
+gulp.task('styles', function () {
```
Comment (matthistuff) [inline gulpfile.js]: Maybe add sourcemaps for development too? (https://www.npmjs.com/package/gulp-less#source-maps)

Comment (cmuecke) [inline gulpfile.js]: To watchers: Nodemon already watches less and js files.

Comment (cmuecke) [inline gulpfile.js]: Regarding gulp in dependencies: Would rather do that because it bloats the installation size.

**Comment by cmuecke** [comment]
@matthistuff We addressed your comments with the 2nd commit.

**Comment by cmuecke** [comment]
@alexander-heimbuch tomorrow we need to manually merge your last notifications commit into this PR. Renaming `src` to `source` breaks the rebase.

**Comment by alexander-heimbuch** [comment]
Rebase done :+1:

**Comment by alexander-heimbuch** [comment]
@matthistuff could you do the review?

**Comment by matthistuff** [comment]
Cool stuff :+1:

**Pull request merged**